### PR TITLE
Added aggregate cells function and dependencies

### PR DESCRIPTION
**Pull request by william-hutchison**
This pull request adds the function aggregate_cells() to the methods file. This addition required multiple functions from the tidybulk package, which I placed in tibybulk_utilities.R. Two further helper functions were required, which I placed in utilities.R.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -41,7 +41,9 @@ S3method(unite,SingleCellExperiment)
 S3method(unnest,tidySingleCellExperiment_nested)
 export("%>%")
 export(add_count)
+export(aggregate_cells)
 export(arrange)
+export(as_matrix)
```
Comment (stemangiola) [inline NAMESPACE:46]: the only exported function should be `aggregate_cells`? Or am I missing something?

**Review by stemangiola — CHANGES_REQUESTED** [review]

**Comment by stemangiola** [comment]
Also, please have a look to the failing tests.

File: R/tidybulk_utilities.R [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    if (!requireNamespace("BiocManager", quietly = TRUE))
+      install.packages("BiocManager", repos = "https://cloud.r-project.org")
+    BiocManager::install("SummarizedExperiment", ask = FALSE)
+  }
+  if (find.package("S4Vectors", quiet = TRUE) %>% length %>% equals(0)) {
+    message("Installing S4Vectors")
+    if (!requireNamespace("BiocManager", quietly = TRUE))
+      install.packages("BiocManager", repos = "https://cloud.r-project.org")
+    BiocManager::install("S4Vectors", ask = FALSE)
+  }
```
Comment (stemangiola) [inline R/tidybulk_utilities.R:71]: These two dependencies can be added to `Imports` in DESCRIPTION, and this code deleted

**Review by stemangiola — CHANGES_REQUESTED** [review]

File: R/tidybulk_utilities.R (hunk 2 above)
Comment (stemangiola) [reply R/tidybulk_utilities.R:71]: This will also solve one WARNING in the tests

File: NAMESPACE (hunk 1 above)
Comment (william-hutchison) [reply NAMESPACE]: Apologies, as_matrix has now been removed from the exported function.

File: R/tidybulk_utilities.R (hunk 2 above)
Comment (william-hutchison) [reply R/tidybulk_utilities.R]: These lines have been removed from the most recent commit. SummarizedExperiment and S4Vectors were already present in the Imports listl

**Comment by stemangiola** [comment]
@william-hutchison congrats! you are now a contributor.

**Pull request merged**